lib/glimesh/api/resolvers/account_resolver.ex

```diff
@@ -29,6 +33,87 @@ defmodule Glimesh.Api.AccountResolver do
     end
   end
 
+  def follow_channel(
+        _parent,
+        %{channel_id: channel_id, has_live_notifications: has_live_notifications},
+        %{context: %{access: access}}
+      ) do
+    with :ok <- Bodyguard.permit(Glimesh.Api.Scopes, :follow, access),
+         %Channel{} = channel <- ChannelLookups.get(channel_id, [:user]),
+         %User{} = streamer <- Accounts.get_user(channel.user.id),
+         %User{} = user <- Accounts.get_user(access.user.id),
+         {:ok, following} <- AccountFollows.follow(streamer, user, has_live_notifications) do
+      {:ok, following}
+    else
+      {:error, %Ecto.Changeset{} = changeset} ->
+        {:error, Api.parse_ecto_changeset_errors(changeset)}
+
+      {:error, message} ->
+        {:error, message}
+
+      nil ->
+        {:error, @error_not_found}
+
+      _ ->
+        {:error, "Unknown error."}
+    end
+  end
+
+  def update_follow(
+        _parent,
+        %{channel_id: channel_id} = args,
+        %{context: %{access: access}}
+      ) do
+    with :ok <- Bodyguard.permit(Glimesh.Api.Scopes, :follow, access),
+         %Channel{} = channel <- Glimesh.ChannelLookups.get(channel_id, [:user]),
+         %Follower{} = follower <- AccountFollows.get_following(channel.user, access.user) do
+      AccountFollows.update_following(follower, args)
+    else
+      {:ok, follower} ->
+        {:ok, follower}
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        {:error, Api.parse_ecto_changeset_errors(changeset)}
+
+      {:error, message} ->
+        {:error, message}
+
+      nil ->
+        {:error, @error_not_found}
+
+      _ ->
+        {:error, "Unknown error."}
+    end
+  end
+
+  def unfollow_channel(
+        _parent,
+        %{channel_id: channel_id},
+        %{context: %{access: access}}
+      ) do
+    with :ok <- Bodyguard.permit(Glimesh.Api.Scopes, :follow, access),
+         %Channel{} = channel when channel != nil <-
+           Glimesh.ChannelLookups.get(channel_id, [:user]),
+         %User{} = streamer when streamer != nil <- Accounts.get_user(channel.user.id),
+         %User{} = user when user != nil <- Accounts.get_user(access.user.id) do
+      AccountFollows.unfollow(streamer, user)
+    else
+      {:ok, following} ->
+        {:ok, following}
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        {:error, Api.parse_ecto_changeset_errors(changeset)}
+
+      {:error, message} ->
```

Comment:
Removed the `is_binary` guard, it was preventing the `:unauthorized` atom returned from `Bodyguard` from being returned. absinthe calls to_string on non-binary values like atoms and handles them just fine.

A different approach would be converting `:unathorized` to a nicer error message, but that's a bigger change and having super pretty errors just isn't the most important thing to do imo, what we have is good enough for now.